packages/lodestar/src/network/gossip/validator.ts

```diff
@@ -117,36 +117,38 @@ export class GossipMessageValidator implements IGossipMessageValidator {
     if (String(subnet) !== getAttestationSubnet(attestation)) {
       return false;
     }
-    // Make sure this is unaggregated attestation
-    if (!this.isUnaggregatedAttestation(attestation)) {
-      return false;
-    }
-    const blockRoot = attestation.data.beaconBlockRoot.valueOf() as Uint8Array;
-    if (!await this.db.block.has(blockRoot) || await this.db.block.isBadBlock(blockRoot)) {
-      return false;
-    }
+    const attestationData = attestation.data;
+    const slot = attestationData.slot;
     const state = await this.db.state.get(this.chain.forkChoice.headStateRoot());
-    if (state.slot < attestation.data.slot) {
-      processSlots(this.config, state, attestation.data.slot);
+    if (state.slot < slot) {
+      processSlots(this.config, state, slot);
     }
     const currentSlot = getCurrentSlot(this.config, state.genesisTime);
-    if (!(attestation.data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= currentSlot &&
-        currentSlot >= attestation.data.slot)) {
+    if (!(slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= currentSlot && currentSlot >= slot)) {
+      return false;
+    }
+    // Make sure this is unaggregated attestation
+    if (!this.isUnaggregatedAttestation(attestation)) {
```

Comment:
I think this should be some standalone utility function so it can be tested properly